src/utils/storage.js

```diff
@@ -0,0 +1,28 @@
+export default ( storageId, defaultValue ) => {
+	const storage = {
+		set: ( query ) => {
+			window.localStorage.setItem( storageId, JSON.stringify( query ) );
+		},
+
+		get: () => {
+			try {
+				return (
+					JSON.parse( window.localStorage.getItem( storageId ) ) ||
+					defaultValue
+				);
+			} catch ( e ) {
+				/* eslint-disable no-console */
+				console.warn(
+					`Unable to parse localstorage property ${ storageId }`,
+					e.message
+				);
+			}
+		},
+
+		reset: () => {
+			storage.set( defaultValue );
+		},
+	};
+
+	return window.localStorage ? storage : null;
```

Comment:
How about putting this check at the function level? I would want `storage( 'id' ).get()` to return `null` instead of fatal error if `window.localStorage` doesn't exist.

Reply:
Yes, that makes sense. Just refactored the `storage` module more along these lines.